src/chainparams.cpp

```diff
@@ -1067,3 +1069,34 @@ void SelectParams(const std::string& network)
 void ClearCheckpoints(CChainParams &params) {
     params.checkpointData = {};
 }
+
+Res UpdateCheckpointsFromFile(CChainParams &params, const std::string &fileName) {
+    std::ifstream file(fileName);
+    if (!file.good()) {
+        return Res::Err("File \"%s\" does not exist.", fileName);
+    }
+
+    ClearCheckpoints(params);
+
+    std::string line;
+    while (std::getline(file, line)) {
```

Comment:
Could you also ignore blank lines, and ones that start with, say `#` for comments?